spec/services/destroy_document_spec.rb

```diff
@@ -0,0 +1,36 @@
+RSpec.describe DestroyDocument do
+  subject { described_class.new(doc) }
+
+  let(:collection) { create(:collection) }
+  let(:user) { create(:user) }
+  let(:file) { fixture_file_upload("gnu_manifesto.md") }
+  let(:filename) { "spec/fixtures/files/gnu_manifesto.md" }
+  let(:link) { nil }
+  let(:chunking_method) { :basic }
+  let(:chunking_profile) { create(:chunking_profile, method: chunking_method, size: 800) }
+  let(:doc) { create(:document, state: :created, link:, file:, filename:, chunking_profile:, collection:, user:) }
+
+  describe "#execute" do
+    before do
+      doc.save
```

Comment:
Not needed.

Reply:
I will stop mentioning these now.

Reply:
When I remove the `doc.save` and `child.save` in these specs, I get test failures:

```
  1) DestroyDocument#execute with childless document destroys document
     Failure/Error: expect { subject.execute }.to change(Document, :count).by(-1)
       expected `Document.count` to have changed by -1, but was changed by 0
     # ./spec/services/destroy_document_spec.rb:24:in `block (4 levels) in <main>'


  2) DestroyDocument#execute when document has children destroys both documents document
     Failure/Error: expect { subject.execute }.to change(Document, :count).by(-2)
       expected `Document.count` to have changed by -2, but was changed by 0
     # ./spec/services/destroy_document_spec.rb:18:in `block (4 levels) in <main>'

 2/2 |============================================================================= 100 =============================================================================>| Time: 00:00:00

Finished in 0.59534 seconds (files took 3.88 seconds to load)
2 examples, 2 failures
```

Reply:
What's happening is that `doc` isn't evaluated until you call `subject.execute` inside the block where you expect it to change the count. So, within the block:

- `doc` is created (+1)
- `child` is created (+1)
- `child` is destroyed (-1)
- `doc` is destroyed (-1)

for a total change of 0. :)

Since `child` will cause `doc` to be evaluated, you could change `let(:child)` to `let!(:child)`.

Reply:
Nice. And using `let!(doc)` got rid of one of the `before` blocks. As for `let(child)` however, Rubocop doesn't like me using `let!`. 😄 

```
spec/services/reset_document_spec.rb:15:7: C: RSpec/LetSetup: Do not use let! to setup objects not referenced in tests.
      let!(:child) { create(:document, parent: doc) }
```

Reply:
@oxaronick, an alternative to `before` block with `child.save`, what do you think about referencing `child` in the example itself? I.e. 

```rb
    context "when document has children" do
      let(:child) { create(:document, parent: doc) }

      it "destroys both documents document" do
        child    # <<<< like so
        expect { subject.execute }.to change(Document, :count).by(-2)
      end
    end
```

Reply:
Does `let!(:child)` not work?

Reply:
Using `let!(child)` works. But `ops lint` doesn't like it.